iroh/src/node.rs

```diff
@@ -653,6 +665,32 @@ impl<D: BaoMap + BaoReadonlyDb, C: CollectionParser> RpcHandler<D, C> {
         })
     }
 
+    fn list_incomplete_blobs(
+        self,
+        _msg: ListIncompleteBlobsRequest,
+    ) -> impl Stream<Item = ListIncompleteBlobsResponse> + Send + 'static {
+        let db = self.inner.db.clone();
+        let local = self.inner.rt.local_pool().clone();
+        futures::stream::iter(db.partial_blobs()).filter_map(move |hash| {
+            let db = db.clone();
+            let t = local.spawn_pinned(move || async move {
+                let Some(entry) = db.get_partial(&hash) else {
+                    return Err(io::Error::new(
+                        io::ErrorKind::NotFound,
+                        "no partial entry found",
+                    ));
+                };
+                io::Result::Ok(ListIncompleteBlobsResponse {
+                    hash,
+                    size: 0,
+                    expected_size: entry.size(),
+                    path: "".to_owned(),
```

Comment:
shouldn't `path` be optional if this is used like this?

Reply:
removed it entirely. incomplete blobs always live in the same place.